x-pack/plugins/enterprise_search/server/routes/workplace_search/groups.test.ts

```diff
@@ -60,16 +59,19 @@ describe('groups routes', () => {
     });
 
     it('creates a request handler', () => {
-      const mockRequest = {
-        body: {
-          group_name: 'group',
-        },
-      };
-      mockRouter.callRoute(mockRequest);
-
       expect(mockRequestHandler.createRequest).toHaveBeenCalledWith({
         path: '/ws/org/groups',
-        ...mockRequest,
+      });
+    });
+
+    describe('validates', () => {
+      it('correctly', () => {
+        const request = {
+          body: {
+            group_name: 'group',
+          },
+        };
+        mockRouter.shouldValidate(request);
```

Comment:
I can see where this misconception came from from routes that were incorrectly passing `body` to `createHandler()`, but the correct place to validate `body` payloads is w/ `mockRouter.shouldValidate`/`mockRouter.shouldThrow`. I went through for all tests w body payloads and created validation tests for them using existing mock data. In one case the validator did indeed catch an instance of incorrect data, so please make sure you're using `shouldValidate` going forward 😄